### PR TITLE
applying new MVVM pattern to sign in view controller and view model

### PR DESCRIPTION
**Pull request by germanStabile**
continuing the work in #94

### CODE REVIEW

**Comment by toptierlabs** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this Pull Request</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: ios-base/View/Onboarding/SignUpViewController.swift [hunk 1]
```diff
@@ -50,14 +49,37 @@ class SignUpViewController: UIViewController {
...
+  
+  func enableSignUpButton(_ enable: Bool) {
+    signUp.alpha = enable ? 1 : 0.5
+    signUp.isEnabled = enable
+  }
+}
+
+extension SignUpViewController: SignUpViewModelDelegate {
+  func formDidChange() {
+    enableSignUpButton(viewModel.hasValidData)
```
Comment (pMalvasio) [inline ios-base/View/Onboarding/SignUpViewController.swift]: Why not just calling `enableSignUpButton ()` and put the `viewModel.hasValidData` inside that method?

Comment (germanStabile) [reply ios-base/View/Onboarding/SignUpViewController.swift]: Because I want to be able to force a value in some cases. i.e: line 27

Comment (pMalvasio) [reply ios-base/View/Onboarding/SignUpViewController.swift]: Every time you force that value is with `false`? Maybe a default `false` there would be nice. (My OCD says that `..(false/true)` is bad for my mental health haha) 🙈

Comment (germanStabile) [reply ios-base/View/Onboarding/SignUpViewController.swift]: implementing agreed solution on #96

**Pull request merged**